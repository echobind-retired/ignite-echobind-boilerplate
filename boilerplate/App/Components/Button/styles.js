import { StyleSheet } from 'react-native';
import { Colors } from '../../Theme';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: Colors.electricBlue,
    padding: 10,
    width: '100%',
    borderRadius: 2,
    justifyContent: 'center',
    alignItems: 'center'
  },
  text: {
    color: Colors.white
  }
});
