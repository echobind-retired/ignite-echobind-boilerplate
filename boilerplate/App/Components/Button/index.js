import React, { Component } from 'react';
import { TouchableOpacity, Text } from 'react-native';
import { Colors } from '../../Theme';

import styles from './styles';

class Button extends Component {
  render() {
    const { text, style } = this.props;

    return (
      <TouchableOpacity {...this.props} style={[styles.container, style]}>
        <Text style={styles.text}>
          {text}
        </Text>
      </TouchableOpacity>
    );
  }
}

export default Button;
