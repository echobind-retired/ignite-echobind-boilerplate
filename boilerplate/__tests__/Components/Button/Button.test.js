import 'react-native';
import React from 'react';
import Button from '../../../App/Components/Button';
import renderer from 'react-test-renderer';

describe('Button', () => {
  describe('text prop', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<Button text="Go" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('additional props', () => {
    it('adds them as props to the underlying TouchableOpacity', () => {
      const tree = renderer.create(<Button what="true" the="heck" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
