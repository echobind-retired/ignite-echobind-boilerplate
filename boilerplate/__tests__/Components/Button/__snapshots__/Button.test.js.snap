// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button additional props adds them as props to the underlying TouchableOpacity 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={undefined}
  />
</View>
`;

exports[`Button text prop renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={undefined}
  >
    Go
  </Text>
</View>
`;
